### ISSUE TITLE
Add flag to pass through exit code of profiled program

### ISSUE THREAD

**Issue by llvtt** [rbspy/rbspy#150]
### Request

I'd like to add a `--passthrough-exit-status` (or some other name) flag to rbspy's `record` subcommand. When this flag is specified, then `rbspy record` would exit with the same status as the command passed to it.

### Use case

I use rbspy to profile a test suite on Circle CI like `rbspy record -- bundle exec rspec ...`. Circle determines build pass/failure based on exit status, but when I run the tests this way, the build succeeds as long as I can collect any profiling data (exit status of rbspy), but I'd like the build to fail if any tests fail (normal exit status behavior of rspec). Because I'm using Circle CI, I can't use `--pid` because that requires the underlying docker container to be run with ptrace enabled (Circle does not do this). While it is possible that I could run a docker container inside another docker container to enable this, but I'd prefer not to add the overhead of nested docker containers if possible.

Thoughts on this feature? I've never written any Rust before, but it seems like this feature could be pretty straightforward, and I'd be willing to take a stab at it.

**Comment by acj** [comment]
Hi, I'm going through old issues and doing a bit of cleanup. Your suggestion is a nice idea and does seem useful for CI situations. Are you still interested in working on this? I think most of the work would be updating the recorder plumbing so that we can pass back the exit code as part of the result. Happy to work with you to figure it out if you're still interested.

**Comment by llvtt** [comment]
@acj Yes, this would still be useful to me, and I am still interested in working on it! I'll look into the code in the next week to see if I can figure it out on my own, but I'll report back if I need any assistance. Thank you!
